circe-yaml-v12/src/main/scala/io/circe/yaml/v12/ParserImpl.scala

```diff
@@ -0,0 +1,137 @@
+package io.circe.yaml.v12
+
+import cats.syntax.either._
+import io.circe._
+import io.circe.yaml.common
+import java.io.{ Reader, StringReader }
+import java.util.Optional
+import org.snakeyaml.engine.v2.api.LoadSettings
+import org.snakeyaml.engine.v2.composer.Composer
+import org.snakeyaml.engine.v2.constructor.StandardConstructor
+import org.snakeyaml.engine.v2.nodes._
+import org.snakeyaml.engine.v2.scanner.StreamReader
+import scala.collection.JavaConverters._
+
+class ParserImpl(settings: LoadSettings) extends common.Parser {
+
+  /**
+   * Parse YAML from the given [[Reader]], returning either [[ParsingFailure]] or [[Json]]
+   * @param yaml
+   * @return
+   */
+  def parse(yaml: Reader): Either[ParsingFailure, Json] = for {
+    parsed <- parseSingle(yaml)
+    json <- yamlToJson(parsed)
+  } yield json
+
+  def parse(yaml: String): Either[ParsingFailure, Json] =
+    parse(new StringReader(yaml))
+
+  def parseDocuments(yaml: Reader): Stream[Either[ParsingFailure, Json]] = parseStream(yaml).map(yamlToJson)
+  def parseDocuments(yaml: String): Stream[Either[ParsingFailure, Json]] = parseDocuments(new StringReader(yaml))
+
+  private[this] def asScala[T](ot: Optional[T]): Option[T] =
+    if (ot.isPresent) Some(ot.get()) else None
+
+  private[this] def createComposer(reader: Reader) =
+    new Composer(settings, new org.snakeyaml.engine.v2.parser.ParserImpl(settings, new StreamReader(settings, reader)))
+
+  private[this] def parseSingle(reader: Reader): Either[ParsingFailure, Node] =
+    Either.catchNonFatal {
+      val composer = createComposer(reader)
+      asScala(composer.getSingleNode)
+    } match {
+      case Left(err)          => Left(ParsingFailure(err.getMessage, err))
+      case Right(None)        => Left(ParsingFailure("no document found", new RuntimeException("no document found")))
+      case Right(Some(value)) => Right(value)
+    }
+
+  private[this] def parseStream(reader: Reader) =
+    createComposer(reader).asScala.toStream
+
+  private[this] object CustomTag {
+    def unapply(tag: Tag): Option[String] = if (!tag.getValue.startsWith(Tag.PREFIX))
+      Some(tag.getValue)
+    else
+      None
+  }
+
+  private[this] class FlatteningConstructor(settings: LoadSettings) extends StandardConstructor(settings) {
+    def flatten(node: MappingNode): MappingNode = {
+      flattenMapping(node)
+      node
+    }
+
+    def construct(node: ScalarNode): Object = super.construct(node) // to make the method public
+  }
+
+  private[this] def yamlToJson(node: Node): Either[ParsingFailure, Json] = {
+    // Isn't thread-safe internally, may hence not be shared
+    val flattener: FlatteningConstructor = new FlatteningConstructor(settings)
+
+    def convertScalarNode(node: ScalarNode) = Either
+      .catchNonFatal(node.getTag match {
+        case Tag.INT if node.getValue.startsWith("0x") || node.getValue.contains("_") =>
```

Comment:
As the snakeyaml-engine readme says, "Only [JSON Schema](http://yaml.org/spec/1.2/spec.html#id2803231) is supported." - that is, it doesn't support any literal formats beyond what's already in JSON. You can inspect the snakeyaml-engine impl of `construct` here and see that it's not doing anything fancier than e.g. `String.toInt`. So we can remove this entire `case Tag.INT if node.getValue ... =>` block - it's not doing anything other than wasting CPU cycles and confusing / misleading future maintainers.
https://bitbucket.org/snakeyaml/snakeyaml-engine/src/master/#markdown-header-snakeyaml-engine-features

Reply:
Do you think this code would be useful once we add Core Schema support to snakeyaml-engine?
https://bitbucket.org/snakeyaml/snakeyaml-engine/issues/38/cannot-parse-integers-encoded-in